### ISSUE TITLE
New download interaction hard to discover

### ISSUE THREAD

**Issue by nadonomy** [vector-im/element-web#18238]
<img width="709" alt="Screenshot 2021-07-26 at 17 26 43" src="https://user-images.githubusercontent.com/4626865/127025466-9fd03f06-85fa-4937-82b4-5f75d57e24ae.png">

The UI on the left is non interactable. I spent a while wondering what was broken. The download interaction in the message action bar is nice, but I think you should be able to download files by just autopiloting to click on the large button/tile in the timeline too.

A simple pragmatic tweak for this would be to:

1. Make the button on the left clickable.
2. Ensure the cursor changes on hover to aid discoverability.
3. Stretch if not difficult: Tooltips.
   1. If the filename doesn't fit on screen, show on immediate hover in one of our tooltips.
   2. Truncate filename (max chars 25-30ish?). Always show file extension.
   3. Show file size, stylising as e.g.: `Filename.ext (20MB)`

Given the entire component is visually stylised as a button, I have confidence in the interaction being discoverable enough for most users. However, we should be on alert to iterate if we get feedback otherwise.

If we learn users benefit from seeing file size in all cases, we should also iterate further on on the design.

**Comment by HarHarLinks** [comment]
In case of an image, "hiding" the download button makes sense, since the primary action is viewing the image (and clicking it to go full screen). Same for video and other media with visual primary interactions.
In case of other files, their visualization is generated by the UI anyway. Things that are not interactive in-place should have download (or perhaps even default-open), as suggested by OP. I would even argue that these app-decided abstract UI things should have a visual cue to "hey click this to download/open", i.e. download button always. This would even apply for e.g. audio (may or may not to voice messages).

This can change when primary actions become something else, e.g. rich inline preview/display of code/plaintext files, however even then a download button should be right there.

In my opinion, the complete loss of what used to be a caption `Filename.ext (size MB) *click to download*` into nothingness and a hidden button is a big 3 steps backwards in the wrong direction.

**Comment by nadonomy** [comment]
@HarHarLinks thanks for the input, totally agree. I've just updated the main issue with some quick/pragmatic fixes we should be able to action to improve this within this release cycle.

@turt2live @dbkr I added a stretch goal on tooltips in 3. If you think this is too large a scope for this release cycle, let me know and I'll file as a separate issue for us to follow up on separately.

(@gaelledel also tagging for visibility; moving quickly/being a bit directive on this so we can act quickly!)

**Comment by HarHarLinks** [comment]
See also: #18197 and more

Here is some similar UI in app.cinny.in
![image](https://user-images.githubusercontent.com/2803622/127492702-3677e837-79d5-4b32-bdf1-681976fc5a9a.png)
This is similar to my suggestion of having a visible download button right on screen, where it is most useful for generic files.

![image](https://user-images.githubusercontent.com/2803622/127492604-2dfaec6f-086b-47cd-8003-8b33b222f567.png)
These are nicer than current stable element, but not the direction current develop is going. Perhaps the bar with info and buttons could instead appear (roll down) on hover?
Mockup with developed as baseline:
![Element Mockup](https://user-images.githubusercontent.com/2803622/127496382-4ac1d596-a371-4f30-9bcc-6607b76caecd.png)

**Comment by turt2live** [comment]
@nadonomy for simplicity, would it be okay to always show the tooltip on hover rather than checking on-screen-ness?

**Comment by nadonomy** [comment]
@HarHarLinks interesting! Please could you file any ideas as a new issue/feature request? We're using this issue to track some pragmatic changes we can action within our current release cycle.

> @nadonomy for simplicity, would it be okay to always show the tooltip on hover rather than checking on-screen-ness?

@turt2live not sure I'm grokking your question exactly. Can you rephrase? I was trying to request/perhaps failing to communicate for us to (1) use our own tooltips, to avoid the lag on showing native HTML ones (2) on hover. Does that match?

**Comment by SimonBrandner** [comment]
> @HarHarLinks interesting! Please could you file any ideas as a new issue/feature request? We're using this issue to track some pragmatic changes we can action within our current release cycle.

It already is a separate issues: https://github.com/vector-im/element-web/issues/18197

**Comment by nadonomy** [comment]
@SimonBrandner edited before your comment, but thx!

**Comment by turt2live** [comment]
@nadonomy essentially doing this to your requirements:
![image](https://user-images.githubusercontent.com/1190097/127507775-8c162c48-65c9-4dfa-aa5d-c918baed7828.png)

😇

**Comment by nadonomy** [comment]
@turt2live ah gotcha. Sorry, thought you were trying to describe an alternate hover interaction related to on-screen-ness!

Yep. I think it might end up feeling oddly redundant. Would also be happy to quickly screen share and rubber duck through how it feels in practise if that would be the most productive.

**Comment by HarHarLinks** [comment]
![image](https://user-images.githubusercontent.com/2803622/127885287-657a70df-80f4-42b3-9487-ecbfab8ba704.png)

nightly 21-08-02

could be slightly better: size unit is cut off in the main UI thing so it's useless, random linebreak in the tooltip.

tbh the tooltip could just always linebreak (right align)
